plugins/browser-plugin-snowplow-ecommerce/src/ua/api.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  trackProductView,
+  trackProductListView,
+  trackProductListClick,
+  trackCheckoutStep,
+  trackTransaction,
+  trackAddToCart,
+  trackRemoveFromCart,
+  trackPromotionClick,
+  trackPromotionView,
+} from '../api';
+import { CheckoutStep } from '../types';
+import {
+  AddToCartEvent,
+  CheckoutStepEvent,
+  ProductDetailEvent,
+  ProductListClickEvent,
+  ProductListViewEvent,
+  PromoClickEvent,
+  PromoViewEvent,
+  PurchaseEvent,
+  RemoveFromCartEvent,
+  UAEnhancedEcommerceObject,
+} from './types';
+
+import { transformUAProductsToSPProducts, transformUAPromotionsToSPPromotions } from './utils';
+
+type Options = {
+  currency?: string;
+};
+
+const DEFAULT_PAYMENT_METHOD = 'unknown';
+
+export function trackEnhancedEcommercePromoView(ecommerce: UAEnhancedEcommerceObject & PromoViewEvent) {
+  const promotions = transformUAPromotionsToSPPromotions(ecommerce.promoView.promotions);
+  /* Enhanced Ecommerce allows multiple promotions to be sent on a single event. */
+  for (let i = 0; i < promotions.length; i++) {
+    trackPromotionView(promotions[i]);
```

Comment:
Think this can just be a `forEach`?

Reply:
Oh derp, will need a wrapper function because of the `trackers` param, this is probably fine, then.